Quantum-Control-Applications-QuAM/Superconducting/README.md

```diff
@@ -0,0 +1,357 @@
+```python
+from qualang_tools.wirer import Connectivity, allocate_wiring
+from qualang_tools.wirer.wirer.channel_specs import mw_fem_spec
+# Define which qubits are present in the system
+qubits = [1, 2, 3, 4, 5, 6]
+# Allocate the wiring to the connectivity object based on the available instruments
+connectivity = Connectivity()
+# Define any custom/hardcoded channel addresses
+q1_res_ch = mw_fem_spec(con=1, slot=1, in_port=1, out_port=1)
+# Single feed-line for reading the resonators & individual qubit drive lines
+connectivity.add_resonator_line(qubits=qubits, constraints=q1_res_ch)
+connectivity.add_qubit_flux_lines(qubits=qubits)
+connectivity.add_qubit_drive_lines(qubits=qubits)
+allocate_wiring(connectivity, instruments)
+```
+3. Finally, the wiring and network information is serialized and store in wiring.json and the QUAM state is initiated in state.json.
+The wiring and port mapping can also be visualized in a matplotlib figure. 
+
+```python
+from quam_libs.quam_builder.machine import build_quam_wiring
+from qualang_tools.wirer import visualize
+# Build the wiring and network into a QuAM machine and save it as "wiring.json"
+build_quam_wiring(connectivity, host_ip, cluster_name, path)
+
+# View wiring schematic
+visualize(connectivity.elements, available_channels=instruments.available_channels)
+```
+![opx1000_wiring](./opx1000_wiring.PNG)
+
+### [2. The QuAM components](./quam_libs/components)
+Describe the structure of the QuAM components and how they can be customized, what to pay attention to...
+Also, how to create a new one and update the wiring and build_quam accordingly.
+
+The hierarchy and structure of QUAM can be detailed as follows: 
+1. [quam_root.py](./quam_libs/components/quam_root.py) represents the highest level in terms of hierarchy.
+It contains the qubits and qubit pairs objects, as well as the wiring, network and Octaves. 
+Its methods are usually applied to all qubits or active qubits. 
+2. The definition of a QuAM transmon is defined in [transmon.py](./quam_libs/components/transmon.py). It contains the
+genral transmon attributes (T1, T2, f_01...) as well as the QuAM components composing the transmon (``xy``, ``resonator`` and 
+`z` in this case). Two-qubit gates can also be implemented by defining a specific qubit pair component as shown in 
+[transmon_pair.py](./quam_libs/components/transmon_pair.py).
+3. The sub-QuAM components are either defined from the base QUAM components directly, such as the qubit xy drive which is
+directly defined as an `IQChannel`, or from user-defined components such as 
+[readout_resonator](./quam_libs/components/readout_resonator.py) or [flux_line](./quam_libs/components/flux_line.py), 
+which allows the customization of their attributes.
+
+
+### [3. Generating the QuAM and state.json](./configuration/make_quam.py)
+Once the QuAM root is implemented with the subsequent components, the QuAM state can be generated automatically and each 
```

Comment:
I do not follow what subsequent components are being referred to here

Reply:
Good point